src/main/java/core/basesyntax/service/impl/ConverterDataToObjectImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ConverterDataToObject;
+
+public class ConverterDataToObjectImpl implements ConverterDataToObject {
+
+    private static final String DATA_SEPARATOR = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public FruitTransaction convertDataToObject(String data) {
+        String[] splitData = data.split(DATA_SEPARATOR);
+        FruitTransaction.Operation operation = identifyOperation(splitData[OPERATION_INDEX]);
+        return new FruitTransaction(operation, splitData[FRUIT_INDEX],
+                Integer.parseInt(splitData[QUANTITY_INDEX]));
```

Comment:
at least two things can go wrong - NumberFormatException for integer parsing, and  IllegalArgumentException for enum parsing. Catch them, throw Runtime with custom message